src/main/java/learnositysdk/request/DataApi.java

```diff
@@ -209,21 +198,41 @@ public void requestRecursive(RequestCallback callback) throws Exception
 
 			if (this.responseHasNext(body)) {
 				this.requestPacket.put("next", meta.get("next"));
-				this.requestString = this.requestPacket.toString();
+				this.updateRequestString();
 				makeNextRequest = true;
 			}
 		} while (makeNextRequest);
 	}
 
-    private JSONObject createResponseObject(Remote remote) throws Exception
-    {
-    	JSONObject response = new JSONObject();
-    	response.put("body", remote.getBody());
-    	response.put("contentType",  remote.getContentType());
-    	response.put("statusCode", remote.getStatusCode());
-    	response.put("timeTaken", remote.getTimeTaken());
-    	return response;
-    }
+	/**
+	 * Sets the request packet, updating the stored string version as well
+	 * @param requestPacket The request packet to store
+	 */
+	private void setRequestPacket(JSONObject requestPacket)
+	{
+		if (Telemetry.isEnabled()) {
+			Telemetry.addToRequest(requestPacket);
+		}
+
+		this.requestPacket = requestPacket;
+		this.updateRequestString();
+	}
+
+	/**
+	 * Updates the stored request string using the current request packet
+	 */
+	private void updateRequestString() {
+		this.requestString = this.requestPacket.toString();
+	}
+
+	private JSONObject createResponseObject(Remote remote) {
```

Comment:
Can this no longer throw an Exception? It used to have throws Exception

Reply:
Many JSON operations no longer throw an exception in the updated version, so there are many cases where our functions no longer throw exceptions thanks to this. 👍